src/test/kotlin/com/terraformation/backend/tracking/db/ObservationScenarioTest.kt

```diff
@@ -0,0 +1,1219 @@
+package com.terraformation.backend.tracking.db
+
+import com.opencsv.CSVReader
+import com.terraformation.backend.RunsAsUser
+import com.terraformation.backend.TestClock
+import com.terraformation.backend.customer.db.ParentStore
+import com.terraformation.backend.customer.model.TerrawareUser
+import com.terraformation.backend.db.DatabaseTest
+import com.terraformation.backend.db.default_schema.OrganizationId
+import com.terraformation.backend.db.default_schema.SpeciesId
+import com.terraformation.backend.db.tracking.MonitoringPlotId
+import com.terraformation.backend.db.tracking.ObservationId
+import com.terraformation.backend.db.tracking.PlantingSiteId
+import com.terraformation.backend.db.tracking.PlantingSubzoneHistoryId
+import com.terraformation.backend.db.tracking.PlantingSubzoneId
+import com.terraformation.backend.db.tracking.PlantingZoneHistoryId
+import com.terraformation.backend.db.tracking.PlantingZoneId
+import com.terraformation.backend.db.tracking.RecordedPlantStatus
+import com.terraformation.backend.db.tracking.RecordedSpeciesCertainty
+import com.terraformation.backend.db.tracking.tables.pojos.RecordedPlantsRow
+import com.terraformation.backend.db.tracking.tables.references.MONITORING_PLOTS
+import com.terraformation.backend.db.tracking.tables.references.OBSERVATION_REQUESTED_SUBZONES
+import com.terraformation.backend.mockUser
+import com.terraformation.backend.point
+import com.terraformation.backend.rectangle
+import com.terraformation.backend.toBigDecimal
+import com.terraformation.backend.tracking.model.ObservationResultsDepth
+import com.terraformation.backend.tracking.model.ObservationResultsModel
+import com.terraformation.backend.tracking.model.ObservationRollupResultsModel
+import com.terraformation.backend.tracking.model.ObservationSpeciesResultsModel
+import com.terraformation.backend.util.calculateAreaHectares
+import io.mockk.every
+import java.io.InputStreamReader
+import java.math.BigDecimal
+import java.nio.file.NoSuchFileException
+import java.time.Instant
+import kotlin.math.sqrt
+import org.jooq.impl.DSL
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.assertAll
+import org.locationtech.jts.geom.MultiPolygon
+
+abstract class ObservationScenarioTest : DatabaseTest(), RunsAsUser {
```

Comment:
I didn't look through this in detail; I assume it's all unmodified from `ObservationResultsStoreTest`.